contracts/MerkleDistributor.sol

```diff
@@ -0,0 +1,115 @@
+pragma solidity 0.6.4;
+
+import "./interface/IBEP20.sol";
+import "./interface/ITokenHub.sol";
+import "./interface/IMerkleDistributor.sol";
+import "./lib/SafeMath.sol";
+import "./System.sol";
+import "./MerkleProof.sol";
+
+contract MerkleDistributor is IMerkleDistributor, System {
+    using SafeMath for uint256;
+
+    string public constant sourceChainID = "Binance-Chain-Tigris"; // TODO: replace with the real chain id
+    address public constant approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa; // TODO: replace with the real address
+    bytes32 public constant override merkleRoot = 0xad4aa415f872123b71db5d447df6bb417fa72c6a41737a82fdb5665e3edaa7c3; // TODO: replace with the real merkle root
+
+    // This is a packed array of booleans.
+    mapping(uint256 => uint256) private claimedBitMap;
+
+    struct ClaimInfo {
+        bytes32 tokenHash;
+        bytes32 node;
+        uint256 index;
+        address contractAddr;
+    }
+
+    function isClaimed(uint256 index) public view override returns (bool) {
+        uint256 claimedWordIndex = index / 256;
+        uint256 claimedBitIndex = index % 256;
+        uint256 claimedWord = claimedBitMap[claimedWordIndex];
+        uint256 mask = (1 << claimedBitIndex);
+        return claimedWord & mask == mask;
+    }
+
+    function _setClaimed(uint256 index) private {
+        uint256 claimedWordIndex = index / 256;
+        uint256 claimedBitIndex = index % 256;
+        claimedBitMap[claimedWordIndex] = claimedBitMap[claimedWordIndex] | (1 << claimedBitIndex);
+    }
+
+
+    function claim(bytes32 tokenSymbol, uint256 tokenIndex, uint256 amount, bytes calldata prefixNode, bytes calldata suffixNode, bytes calldata ownerSignature, bytes calldata approvalSignature, bytes32[] calldata merkleProof) external override {
+        ClaimInfo memory claimInfo;
+        claimInfo.tokenHash = keccak256(abi.encodePacked(tokenIndex, tokenSymbol, amount));
+        claimInfo.node = keccak256(abi.encodePacked(prefixNode, claimInfo.tokenHash, suffixNode));
```

Comment:
I think it is a dangerous concat here, because `prefixNode` and  `suffixNode` has no particular meaning, I can get any part of node and explain it.

Reply:
I refactored this

Currently the Merkel leaf node is changed to use `address, tokenIndex, tokenSymbol, amount` as a hash

This can ensure the length of calldata and avoid the problem you mentioned.